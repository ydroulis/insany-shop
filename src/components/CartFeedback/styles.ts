"use client"

import styled from 'styled-components'

export const Container = styled.div<{ $color: string; $background: string }>`
    display: flex;
    align-items: center;
    justify-content: center;

    width: 300px;
    height: 60px;

    border-radius: 8px;

    background-color: ${({ $background }) => $background};

    color: ${({ $color }) => $color};

    position: fixed;
    bottom: 50px;
    right: 50px;

    z-index: 999;
`