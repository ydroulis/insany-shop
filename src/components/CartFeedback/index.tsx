import React from 'react';
import * as S from './styles';

interface CartFeedbackProps {
    children: React.ReactNode;
    color: string;
    background: string;
    type?: 'success' | 'error' | 'info';
}

const CartFeedback: React.FC<CartFeedbackProps> = ({
    children,
    color,
    background,
    type = 'info'
}) => {
    const ariaLabelMap = {
        success: 'Aviso de sucesso',
        error: 'Aviso de erro',
        info: 'Aviso de informação'
    };

    return (
        <S.Container
            $color={color}
            $background={background}
            role="alert"
            aria-live="assertive"
            aria-label={ariaLabelMap[type]}
        >
            <p>{children}</p>
        </S.Container>
    );
};

export default CartFeedback;
