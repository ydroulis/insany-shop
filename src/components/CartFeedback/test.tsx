import React from 'react';
import { render, screen } from '@testing-library/react';
import CartFeedback from './';

// Tipagem do mock do Container
interface MockContainerProps extends React.HTMLAttributes<HTMLDivElement> {
    children: React.ReactNode;
    $color?: string;
    $background?: string;
    'aria-live'?: 'off' | 'polite' | 'assertive';
}

// Mock do styled-component Container
jest.mock('./styles', () => ({
    Container: ({ children, $color, $background, ...props }: MockContainerProps) => (
        <div
            data-testid="container"
            data-color={$color}
            data-background={$background}
            {...props}
        >
            {children}
        </div>
    ),
}));

describe('<CartFeedback />', () => {
    it('should render children propperly', () => {
        render(
            <CartFeedback color="#000" background="#fff">
                Produto adicionado ao carrinho
            </CartFeedback>
        );

        expect(
            screen.getByText(/produto adicionado ao carrinho/i)
        ).toBeInTheDocument();
    });

    it('should have role alert and aria-live assertive', () => {
        render(
            <CartFeedback color="#000" background="#fff">
                Mensagem de teste
            </CartFeedback>
        );

        const container = screen.getByTestId('container');
        expect(container).toHaveAttribute('role', 'alert');
        expect(container).toHaveAttribute('aria-live', 'assertive');
    });

    it('should apply aria-label propperly for success', () => {
        render(
            <CartFeedback color="#000" background="#fff" type="success">
                Sucesso
            </CartFeedback>
        );

        const container = screen.getByTestId('container');
        expect(container).toHaveAttribute('aria-label', 'Aviso de sucesso');
    });

    it('should apply aria-label propperly for error', () => {
        render(
            <CartFeedback color="#000" background="#fff" type="error">
                Erro
            </CartFeedback>
        );

        const container = screen.getByTestId('container');
        expect(container).toHaveAttribute('aria-label', 'Aviso de erro');
    });

    it('should apply aria-label propperly for info (default)', () => {
        render(
            <CartFeedback color="#000" background="#fff">
                Info
            </CartFeedback>
        );

        const container = screen.getByTestId('container');
        expect(container).toHaveAttribute('aria-label', 'Aviso de informação');
    });

    it('should pass color and background to container', () => {
        render(
            <CartFeedback color="red" background="blue">
                Testando estilos
            </CartFeedback>
        );

        const container = screen.getByTestId('container');
        expect(container).toHaveAttribute('data-color', 'red');
        expect(container).toHaveAttribute('data-background', 'blue');
    });

    it('should match snapshot', () => {
        const { container } = render(
            <CartFeedback color="red" background="blue">
                Testando estilos
            </CartFeedback>
        );
        expect(container).toMatchSnapshot();
    });
});
