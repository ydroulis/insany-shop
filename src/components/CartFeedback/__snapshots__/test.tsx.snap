// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<CartFeedback /> should match snapshot 1`] = `
<div>
  <div
    aria-label="Aviso de informação"
    aria-live="assertive"
    data-background="blue"
    data-color="red"
    data-testid="container"
    role="alert"
  >
    <p>
      Testando estilos
    </p>
  </div>
</div>
`;
